src/polyswarm_api/__init__.py

```diff
@@ -500,11 +551,111 @@ def _reveal_closed(self, result):
         return all(('window_closed' in file and file['window_closed']) or ('failed' in file and file['failed'])
                    for file in result['files'])
 
+    async def _new_hunt(self, rules, scan_type):
+        """
+        Create a new scan, either live or historical.
+
+        :param rules: String containing YARA rules to user
+        :param scan_type: Type of scan, "live" or "historical"
+        :return: ID of the new scan
+        """
+        data = {"yara": rules}
+
+        async with self.post_semaphore:
+            logger.debug(f"Posting rules with api-key {self.api_key}")
+            async with aiohttp.ClientSession() as session:
+                try:
+                    async with session.post(f"{self.hunt_uri}/{scan_type}", json=data,
+                                            headers={"Authorization": self.api_key}) as raw_response:
+                        try:
+                            response = await raw_response.json()
+                        except Exception:
+                            response = await raw_response.read() if raw_response else 'None'
+                            logger.error(f'Received non-json response from PolySwarm API: {response}')
+                            response = {"result": "error"}
+                        if raw_response.status // 100 != 2:
+                            errors = response.get('errors')
+                            logger.error(f"Error posting to PolySwarm API: {errors}")
+                            response = {"status": "error"}
+                        return response
+                except Exception:
+                    logger.error('Server request failed')
+                    return {'status': "error"}
+
+    async def new_live_hunt(self, rules):
+        """
+        Create a new live scan, and replace the currently running YARA rules.
+
+        :param rules: String containing YARA rules to install
+        :return: ID of the new scan.
+        """
+        return await self._new_hunt(rules, "live")
+
+    async def new_historical_hunt(self, rules):
+        """
+        Run a new historical scan.
+
+        :param rules: String containing YARA rules to install
+        :return: ID of the new scan.
+        """
+        return await self._new_hunt(rules, "historical")
+
+    async def _get_hunt_results(self, rule_id=None, scan_type="live"):
+        """
+
+        :param rule_id: Rule ID (None if latest rule results are desired)
+        :param scan_type: Type of scan, "live" or "historical"
+        :return: Matches to the rules
+        """
+
+        params = {"results": "true"}
```

Comment:
Why a `results` param in addition to GETing `.../results`? Is one of these superfluous?

Reply:
That's old cruft that isn't needed, will remove.